userspace/libscap/engine/savefile/scap_savefile.c

```diff
@@ -2095,7 +2088,7 @@ static int32_t init(struct scap* main_handle, struct scap_open_args* args)
 
 	if(res != SCAP_SUCCESS)
 	{
-		scap_reader_close(reader);
+		reader->close(reader);

```

Comment:
This looks kinda weird, maybe the close function needs to be separated from the reader, similar to the open functions. Also, it'd be nice to not leave freeing `reader` to the caller when `open` reserves it for us, seems like a good way to get unwanted memory leaks.